### ISSUE TITLE
Snippet support

### ISSUE THREAD

**Issue by ghost** [ul/kak-lsp#678]
I have been looking into kak-lsp's snippet support. From the docs it seems like there should be support for snippets: https://github.com/kak-lsp/kak-lsp#snippets. In my configuration I've enabled snippet support with `snippet_support = true`.

Now I've tested this with the typescript language server in a javascript file. The ts language server is working as expected. I would expect common patterns like a for loop to be completed as a snippet, as can be seen in the vscode docs here for example: https://code.visualstudio.com/docs/editor/userdefinedsnippets#_builtin-snippets. But that does not work with kak-lsp out of the box, all that happens is that the language server in kakoune suggests completing the `for` keyword.

The documentation on kak-lsp's snippet support is a little sparse. Should there be autocompletion via snippets for common language features like a for loop, a switch, etcetera? Does the user need to install these snippets themselves?

**Comment by krobelus** [comment]
yeah snippets are supported (see 22b0602f58ca2815c76ed35baef190ef6eeddc93 for how they usually work) but we don't have VSCode's extra user-defined snippets. They need to come from the language server. With rust-analyzer there are lots of snippets.

user-defined snippets could go into a dedicated snippets plugin

**Comment by krobelus** [comment]
Ideally the the language server supplies snippets, then it works for all editors.

it's possible to add a new completer option that works like lsp_completions.
It's probably not ergonomic because you'd need to match a few implementation details.

**Comment by ZakharEl** [comment]
https://github.com/lunjon/snippetls is a good candidate for custom snippet support
